### PR TITLE
more formats supported

### PR DESCRIPTION
**Pull request by hoodie**
Hi there,
after the [benchmarks](http://github.com/hoodie/dateparser_benchmarks) I did received a little bit of attention I decided it would be nice to use your parse implementation in [datetime](http://github.com/ogham/rust-datetime/). In order to do that I added support for further formats to the parser. If you sign off on these changes it would be great if we could also release this crate on crates.io.

Things I did:
1. remove `easy.rs`
2. remove `chrono` dependencies ( this should only parse, validation should be done by a complete datetime library)
3. add support for ordinal and week formats

thanks for reading this

### CODE REVIEW

**Comment by badboy** [comment]
Thank you, I'll check that next week.

**Comment by hoodie** [comment]
cool

File: src/lib.rs [hunk 1]
```diff
@@ -1,177 +1,90 @@
+#![allow(dead_code)]
```
Comment (badboy) [inline src/lib.rs]: Why allow dead code?

File: src/lib.rs [hunk 2]
```diff
@@ -1,177 +1,90 @@
...
 
-#[derive(Clone,Eq,PartialEq)]
+#[derive(Clone,Eq,PartialEq,Debug)]
 pub struct Time {
     pub hour: u32,
     pub minute: u32,
     pub second: u32,
-    pub tz_offset: i32,
+    pub millisecond: u32,
+    pub tz_offset: (i32,i32),
```
Comment (badboy) [inline src/lib.rs]: Why the tuple? Might be good to document that.

File: src/macros.rs [hunk 3]
```diff
@@ -50,6 +52,179 @@ macro_rules! char_between(
...
+named!(upper_hour <u32>, chain!(tag!("2") ~ s:char_between!('0','4') , || 20+buf_to_u32(s)));
+named!(pub hour <u32>, alt!(lower_hour | upper_hour));
+
+// MM
+named!(below_sixty <u32>, chain!(f:char_between!('0','5') ~ s:char_between!('0','9'), || { buf_to_u32(f)*10 + buf_to_u32(s) } ));
+named!(upto_sixty <u32>, alt!(below_sixty | map!(tag!("60"), |_| 60)));
+
+named!(pub minute <u32>, call!(below_sixty));
+named!(pub second <u32>, call!(upto_sixty));
+named!(pub millisecond <u32>, chain!( ms: is_a!("0123456789"), || buf_to_u32(ms) ) );
```
Comment (badboy) [inline src/macros.rs]: I thinkg this can be `map!` instead of `chain!`

File: src/lib.rs (hunk 1 above)
Comment (hoodie) [inline src/lib.rs]: I'm still not completely done with everthing, so thats a WIP artifact

File: src/macros.rs (hunk 3 above)
Comment (hoodie) [inline src/macros.rs]: thanks

File: src/lib.rs (hunk 1 above)
Comment (badboy) [inline src/lib.rs]: :) Perfect! Just leave a comment once it's out of WIP.
I like it so far and turning this into a proper crate would be nice.

**Comment by hoodie** [comment]
Ok, I think I'm there

What happens from now on I'd like to leave up to you.
One thing that is still missing is a nice `Error` implementation. What do you think?

**Comment by badboy** [comment]
Thanks! I really appreciate all that work. It's gonna take me a bit to read this, but considering we have quite some tests I guess it's pretty solid already, so I'll end up merging it ASAP.

After that I definitely will roll it out as a crate and then we can figure out how to properly handle errors.

**Comment by badboy** [comment]
@homu r+

**Comment by badboy** [comment]
@homu r+

**Pull request merged**